be/src/http/action/compaction_action.cpp

```diff
@@ -66,17 +76,102 @@ Status CompactionAction::_handle_show_compaction(HttpRequest* req, std::string*
     return Status::OK();
 }
 
+Status get_params(HttpRequest* req, uint64_t* tablet_id, uint32_t* schema_hash) {
+    std::string req_tablet_id = req->param(TABLET_ID_KEY);
+    std::string req_schema_hash = req->param(TABLET_SCHEMA_HASH_KEY);
+
+    try {
+        *tablet_id = std::stoull(req_tablet_id);
+        *schema_hash = std::stoul(req_schema_hash);
+    } catch (const std::exception& e) {
+        std::string msg = fmt::format("invalid argument.tablet_id:{}, schema_hash:{}", req_tablet_id, req_schema_hash);
+        LOG(WARNING) << msg;
+        return Status::InvalidArgument(msg);
+    }
+
+    return Status::OK();
+}
+
+Status CompactionAction::_handle_run_compaction(HttpRequest* req, std::string* json_result) {
+    if (_running) {
+        return Status::TooManyTasks("Manual compaction task is running");
+    }
+    _running = true;
+    uint64_t tablet_id;
+    uint32_t schema_hash;
+    RETURN_IF_ERROR(get_params(req, &tablet_id, &schema_hash));
+
+    TabletSharedPtr tablet = StorageEngine::instance()->tablet_manager()->get_tablet(tablet_id, schema_hash);
+    RETURN_IF(tablet == nullptr,
+              Status::InvalidArgument(fmt::format("Not Found table:{}, schema hash:{}", tablet_id, schema_hash)));
+
+    std::string compaction_type = req->param(PARAM_COMPACTION_TYPE);
+    if (compaction_type != PARAM_COMPACTION_BASE && compaction_type != PARAM_COMPACTION_CUMULATIVE) {
+        return Status::NotSupported(strings::Substitute("The compaction type '$0' is not supported", compaction_type));
+    }
+
+    StarRocksMetrics::instance()->cumulative_compaction_request_total.increment(1);
+
+    auto* mem_tracker = ExecEnv::GetInstance()->compaction_mem_tracker();
+
+    if (compaction_type == PARAM_COMPACTION_CUMULATIVE) {
```

Comment:
There are already CompationType defined storage/olap_common.h
It's better to reuse it.